### PR TITLE
Add Tekton target documentation

### PR DESCRIPTION
**Pull request by cab105**
The original Tekton user documentation was ported over from the development repo assuming the target would run on-prem.  This guide provides details on how to create a task, how to create the target, and what the event should look like.

### CODE REVIEW

File: docs/targets/index.md [hunk 1]
```diff
@@ -10,20 +10,15 @@ The following is a list of Triggermesh event destinations known as `Targets`. So
     * [AWS SQS](./awssqs.md): Forward event payload to AWS Simple Queueing Service (SQS).
     * [AWS Kinesis](./awskinesis.md): Forward event payload to AWS Kinesis.
     * [AWS EventBridge](./awseventbridge.md): Forward arbitrary events to AWS EventBridge.
+* [Confluent](./confluent.md): Forward events to Confluent Kafka.
 * [Datadog](./datadog.md): Forward event payload to Datadog.
 * [Elasticsearch](./elasticsearch.md): Send events to Elasticsarch to be indexed.
+* [Googlesheets](./googlesheets.md): Send events to a GoogleSheets.
```
Comment (antoineco) [inline docs/targets/index.md:16]: Are we certain we want to show [this](https://docs.triggermesh.io/targets/googlesheets/) to the public?

File: docs/targets/index.md [hunk 2]
```diff
@@ -10,20 +10,15 @@ The following is a list of Triggermesh event destinations known as `Targets`. So
     * [AWS SQS](./awssqs.md): Forward event payload to AWS Simple Queueing Service (SQS).
     * [AWS Kinesis](./awskinesis.md): Forward event payload to AWS Kinesis.
     * [AWS EventBridge](./awseventbridge.md): Forward arbitrary events to AWS EventBridge.
+* [Confluent](./confluent.md): Forward events to Confluent Kafka.
```
Comment (antoineco) [inline docs/targets/index.md:13]: Are we certain we want to show [this](https://docs.triggermesh.io/targets/confluent/) to the public?

File: docs/targets/tekton.md [hunk 3]
```diff
@@ -1,85 +1,75 @@
 # Tekton Pipeline Event Target for Knative Eventing
```
Comment (antoineco) [inline docs/targets/tekton.md:1]:
```suggestion
# Event Target for Tekton Pipeline
```

File: docs/targets/tekton.md [hunk 4]
```diff
@@ -1,85 +1,75 @@
...
 
-This event target integrates with Tekton pipelines using received Cloud Event
-messages to trigger a pipeline run or a task run.
+This event target receives [CloudEvents][ce] over HTTP and sends them to [Tekton][tekton]
+to invoke a Tekton build task.
```
Comment (antoineco) [inline docs/targets/tekton.md:4]: This target is a "integration target" so we don't really "send" stuff to Tekton, but we rather process tasks in the shape of events.

Maybe we should rephrase this slightly to "receive CloudEvents that represent tasks to trigger Tekton builds."?

File: docs/targets/tekton.md [hunk 5]
```diff
@@ -1,85 +1,75 @@
...
+| **buildType**| string  |  The run object type consisting of `task` or `pipeline` |
+|  **name** |  string | The Tekton Task or Pipeline object to invoke  |
+| **params**| map[string]string | Dictionary mapping of parameters to pass to the Tekton Task or Pipeline|
+
+No response events are created with this Target type.
+
+---
+**NOTE:**
+TaskRun and PipelineRun objects nor their associated pods are deleted after execution.
+It is up to the user do perform the clean-up.
```
Comment (antoineco) [inline docs/targets/tekton.md:70]:
```suggestion
It is up to the user to perform the clean-up.
```

**Review by antoineco — APPROVED** [review]
Looking good. Only 3 minor comments and we probably need to disable the links to Confluent and Google Sheets.

File: docs/targets/index.md (hunk 1 above)
Comment (cab105) [reply docs/targets/index.md:16]: I am cleaning this up, and will be included in the next PR.

File: docs/targets/index.md (hunk 2 above)
Comment (cab105) [reply docs/targets/index.md:13]: I am cleaning this up, and it will be updated in the next PR.

Comment (antoineco) [reply docs/targets/index.md:13]: Maybe this diff belongs in the "next PR" then?

File: docs/targets/tekton.md (hunk 5 above)
Comment (cab105) [reply docs/targets/tekton.md:70]: Doh! Thank you for the catch!

File: docs/targets/tekton.md (hunk 3 above)
Comment (cab105) [reply docs/targets/tekton.md:1]: Thanks that does look cleaner.

File: docs/targets/tekton.md (hunk 4 above)
Comment (cab105) [reply docs/targets/tekton.md:4]: How about instead:
`This target receives a [CloudEvents][ce] over HTTP and will use it to create a [Tekton][tekton] TaskRun or PipelineRun object.`

Comment (antoineco) [reply docs/targets/tekton.md]: Sounds much better than my clumsy suggestion!

**Pull request merged**